content/blog/dotnet-preview/index.md

```diff
@@ -0,0 +1,238 @@
+
+Millions of . NET developers can now use their favorite languages and open source ecosystems to build modern, cloud native applications. We've added support for C#, F# and Visual Basic. Because the . NET framework is available on Windows, Linux and macOS you have a choice of platforms to use. You can create cloud resources by writing **Microsoft . NET Core** programs to build and deploy AKS Clusters, Functions, Azure App Services, Virtual Machines, Cosmos DBs, KeyVaults, and much, much more.
+
+## What Does Pulumi Enable?
+
+As a . NET developer you can:
+
+* Declare infrastructure in C#, F#, or Visual Basic NET.
+* Code with favorite IDEs, including Visual Studio, Visual Studio Code, and Rider, with auto-completion and refactoring tools.
+* Catch mistakes early on with red markers and build errors.
+* Allow package managers to reference any NuGet library compatible with . NET Core 3.0.
+* Use the Pulumi CLI or CI/CD integrations to perform deployments predictably and reliably.
+* Maintain your infrastructure in a repository and version it
+* Build modern, reliable, and scalable applications using a multitude of cloud architectures, including containers, serverless functions, and static websites.
+
+You can do all of this without having to learn a DSL or script CLI commands. With Pulumi, developers and operators have a shared foundation for building modern applications.
+
+## How Pulumi . NET Works
+
+Pulumi is declarative even though it uses general purpose programming languages. It does this by declaring a set of resources and the Pulumi engine orchestrates the CRUD operations to build and deploy infrastructure.
+
+We use a . NET Core console application to declare infrastructure. Let's create an Azure App Service with a SQL database. The first step is make sure that your environment meets the prerequisites:
+
+### Prerequisites
+
+1. [Install Pulumi](https://www.pulumi.com/docs/get-started/install/)
+
+1. [Install . NET Core 3.0+](https://dotnet.microsoft.com/download)
+
+1. [A Microsoft Azure account](https://azure.microsoft.com/en-us/free/)
+
+1. [Install Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)
+
+### Get the Examples
+
+Pulumi provides a repository of examples to get you started. Clone the examples from GitHub.
+
+``` bash
+git clone https://github.com/pulumi/examples/
+cd examples/
+```
+
+### Create a New Stack
+
+Change the directory to `azure-cs-appservice` directory
+
+``` bash
+$ cd azure-cs-appservice
+```
+
+There are four files in the Pulumi project, the `Azure.Appservice.csproj` which sets the . NET SDK and the Pulumi packages for the project. `Pulumi.yaml` holds project parameters such as name, runtime and the database password. `SharedAccessSignature.cs` specifies the storage to hold code, which is the `wwwroot` directory in this project. `Program.cs` is the main program for declaring resources.
```

Comment:
`SharedAccessSignature` contains a helper function to generate storage blob access URLs with SAS tokens in them. It's a demo that you can do any kind of custom code inside your Pulumi program. It doesn't have much to do with the `wwwroot` folder.

Reply:
rephrased it